### ISSUE TITLE
Win 10 10.0.19041.1023 not supported

### ISSUE THREAD

**Issue by Pavel-Shift** [stascorp/rdpwrap#1420]

**Comment by guweigang** [comment]
+1

**Comment by sebaxakerhtc** [comment]
https://github.com/sebaxakerhtc/rdpwrap.ini

**Comment by dmcdivitt** [comment]
works for me, thanks

**Comment by dmcdivitt** [comment]
[rdpwrap.ini.txt](https://github.com/stascorp/rdpwrap/files/6604950/rdpwrap.ini.txt)

derived from @sebaxakerhtc above

**Comment by StarfighterJ** [comment]
OK :) Autoupdate did work for me, wow This was the first time I did not have to uninstall and reinstall all of it

**Comment by guweigang** [comment]
works for me

**Comment by sebaxakerhtc** [comment]
> _No description provided._

Close issue, please

**Comment by remmahad** [comment]
Worked also for 10.0.19041.964

**Comment by sebaxakerhtc** [comment]
> Worked also for 10.0.19041.964

Of course. It's a older one

**Comment by remmahad** [comment]
> > Worked also for 10.0.19041.964
> 
> Of course. It's a older one

Yeah sure! But now you can search for that specific version on google and hopefully can find this post :)
Is just a note.

**Comment by sebaxakerhtc** [comment]
@tryfail 
Why this reaction? Does it not work for you?

**Comment by gamma-577** [comment]
@sebaxakerhtc 


10 10.0.19041.1023 offsets works like a charm! Thank you!!

-Gamma


edit : typo

**Comment by CrazyBebop** [comment]
How do you apply the new rdpwrap.ini when I click update it says 2018 version already installed, I put the rdpwrap.ini in the same folder as the installer, but the installer continues to install the older version.

Thanks.

**Comment by SamuFr** [comment]
I updated Rdpwrap.ini with Autoupdater from Asmtron.
With Tetmsrv.dll 32 bits 10.0.19041.1023 I have "listening" and "supported" but it is not possible to open more than one session (if a session is already open locally, I get "Another user is signed in ... ").
It is likely that Rdpwrap.ini has a wrong address in the x86 section ...

Who would have the solution ?

**Comment by dmcdivitt** [comment]
When I replace the ini file and RDPWrapper still doesn't work I run the install bat file, then replace the ini file again.


[RDPWinst_uninstall.bat.txt](https://github.com/stascorp/rdpwrap/files/6642715/RDPWinst_uninstall.bat.txt)
[rdpwrap_update.bat.txt](https://github.com/stascorp/rdpwrap/files/6642716/rdpwrap_update.bat.txt)
[RDPWinst_install.bat.txt](https://github.com/stascorp/rdpwrap/files/6642717/RDPWinst_install.bat.txt)

**Comment by SamuFr** [comment]
Has this last answer anything to do with my question ?

**Comment by contras2k** [comment]
Both x86 and x64 sections 

[10.0.19041.1023]
LocalOnlyPatch.x86    =1
LocalOnlyOffset.x86   =B6419
LocalOnlyCode.x86     =jmpshort
SingleUserPatch.x86   =1
SingleUserOffset.x86  =3BD35
SingleUserCode.x86    =nop
DefPolicyPatch.x86    =1
DefPolicyOffset.x86   =3E8A9
DefPolicyCode.x86     =CDefPolicy_Query_eax_ecx
SLInitHook.x86        =1
SLInitOffset.x86      =68598
SLInitFunc.x86        =New_CSLQuery_Initialize

LocalOnlyPatch.x64    =1
LocalOnlyOffset.x64   =89C51
LocalOnlyCode.x64     =jmpshort
SingleUserPatch.x64   =1
SingleUserOffset.x64  =0CB22
SingleUserCode.x64    =Zero
DefPolicyPatch.x64    =1
DefPolicyOffset.x64   =18A15
DefPolicyCode.x64     =CDefPolicy_Query_eax_rcx
SLInitHook.x64        =1
SLInitOffset.x64      =1DFDC
SLInitFunc.x64        =New_CSLQuery_Initialize

[10.0.19041.1023-SLInit]
bServerSku.x86        =D1A08
bRemoteConnAllowed.x86=D1A1C
bFUSEnabled.x86       =D1A28
bAppServerAllowed.x86 =D1A14
bMultimonAllowed.x86  =D1A20
lMaxUserSessions.x86  =D1A0C
ulMaxDebugSessions.x86=D1A24
bInitialized.x86      =D1A04

bInitialized.x64      =107108
bServerSku.x64        =10710C
lMaxUserSessions.x64  =107110
bAppServerAllowed.x64 =107118
bRemoteConnAllowed.x64=107120
bMultimonAllowed.x64  =107124
ulMaxDebugSessions.x64=107128
bFUSEnabled.x64       =10712C

**Comment by SamuFr** [comment]
Thank you so much, Contras2k, for the code. Now it's OK with Win 32 bits.
In fact, the whole 32-bit sections for 10.0.19041.1023 were missing.
I don't understand why "listening" and "supported" could be green without these sections...

**Comment by dionelfer** [comment]
hay tiene el ini. para la version HOME?

**Comment by dionelfer** [comment]
alguien tiene la 10.0.19041.1023 para la version Windows Home single lenguaje de 64 bits..??

**Comment by dcbasso** [comment]
does not work for me...
Always show "NOT SUPPORTED"

**Comment by contras2k** [comment]
> I don't understand why "listening" and "supported" could be green without these sections...

Actually the patcher has no meaning to check if patching was made correctly or if required effects were achieved. It doesn't know original codes, only addresses. If termsrv.dll is big enough to contain these addresses in executable area the patcher simply puts new codes and hopes that everything will be fine from now on. And it's up to you to check if new functionality is really unlocked.

**Comment by dionelfer** [comment]
> alguien tiene la 10.0.19041.1023 para la versión **Windows Home single lenguaje de 64 bit**s .. ??

Encontre la solucion por otro lugar. dejo el codigo .ini 

[10.0.19041.1023]
LocalOnlyPatch.x64 = 1
LocalOnlyOffset.x64 = 89C51
LocalOnlyCode.x64 = jmpshort
SingleUserPatch.x64 = 1
SingleUserOffset.x64 = 0CB22
SingleUserCode.x64 = Cero
DefPolicyPatch.x64 = 1
DefPolicyOffset.x64 = 18A15
DefPolicyCode.x64 = CDefPolicy_Query_eax_rcx
SLInitHook.x64 = 1
SLInitOffset.x64 = 1DFDC
SLInitFunc.x64 = New_CSLQuery_Initialize

[SLInit]
bServerSku = 1
bRemoteConnAllowed = 1
bFUSEnabled = 1
bAppServerAllowed = 1
bMultimonAllowed = 1
lMaxUserSessions = 0
ulMaxDebugSessions = 0
bInicializado = 1

[10.0.19041.1023-SLInit]
bInitialized.x64       = 107108
bServerSku.x64         = 10710C
lMaxUserSessions.x64   = 107110
bAppServerAllowed.x64 = 107118
bRemoteConnAllowed.x64 = 107120
bMultimonAllowed.x64   = 107124
ulMaxDebugSessions.x64 = 107128
bFUSEnabled.x64        = 10712C

Probado y funcionando con soporte multi-usuario.

**Comment by dcbasso** [comment]
Nothing!

**Comment by dcbasso** [comment]
Using dll from #1426 and file ini from here and NOT SUPPORTED. @sebaxakerhtc can help?

**Comment by sebaxakerhtc** [comment]
> Using dll from #1426 and file ini from here and NOT SUPPORTED. @sebaxakerhtc can help?

It's not clear for me what you doing...
All works fine on x64 with original termsrv.dll and my modded rdpwrapper

**Comment by Johncooool** [comment]
> does not work for me...
> Always show "NOT SUPPORTED"

Make sure that the Autoupdate is bypassing the firewall.

I had the same issue and solved it by running the Autopdate as admin from C:\Program Files\RDP Wrapper or the directory where it is installed.

**Comment by XoRoNioX** [comment]
Help - see post [https://github.com/stascorp/rdpwrap/issues/1436](url)  Fully supported but cannot multi login -

**Comment by lgzcoollg** [comment]
when the states is fully supported,but no listening, this link could solve!!!  
https://github.com/stascorp/rdpwrap/issues/1438#issuecomment-862981696

**Comment by robertodan** [comment]
> Try:
> 
>     1. Download RDP Wrapper 1.6.2 and extract this to any directory.
> 
>     2. Place `rdpwrap.ini` in the same directory.
> 
>     3. Open CMD as administrator
> 
>     4. Execute this commands
> 
> 
> ```
> cd C:\path\to\directory\
> rdpwinst -u -k
> rdpwinst -i
> ```
> 
>     5. Enjoy!
> 
> 
> Original message: https://t.me/rdpwrap/49131


I get "Access is denied"
